cpp/src/arrow/filesystem/gcsfs_test.cc

```diff
@@ -352,6 +360,68 @@ TEST_F(GcsIntegrationTest, ReadObjectInfoInvalid) {
   EXPECT_EQ(result.status().code(), StatusCode::IOError);
 }
 
```

Comment:
Is it also possible to add a simple roundtrip test between `ToObjectMetadata` and `FromObjectMetadata`?

Reply:
Done (with some limitations, not all fields have public modifiers in `gcs::ObjectMetadata`)